### ISSUE TITLE
TestRoundTripTypes should also test extensions

### ISSUE THREAD

**Issue by mml** [kubernetes/kubernetes#18609]
AFAICT, it only tests the v1 API objects.

```console
% GOPATH=$(pwd)/Godeps/_workspace:$GOPATH go test -v k8s.io/kubernetes/pkg/api -run TestRoundTripTypes | grep -Po 'working on \K.*' | sort | column -c80
APIGroup in                     PersistentVolumeList in 
APIGroupList in                 PodAttachOptions in 
APIResourceList in              PodExecOptions in 
APIVersions in                  Pod in 
Binding in                      PodList in 
ComponentStatus in              PodLogOptions in 
ComponentStatusList in          PodProxyOptions in 
DeleteOptions in                PodStatusResult in 
Endpoints in                    PodTemplate in 
EndpointsList in                PodTemplateList in 
Event in                        RangeAllocation in 
EventList in                    ReplicationController in 
LimitRange in                   ReplicationControllerList in 
LimitRangeList in               ResourceQuota in 
List in                         ResourceQuotaList in 
ListOptions in                  Secret in 
Namespace in                    SecretList in 
NamespaceList in                SerializedReference in 
Node in                         ServiceAccount in 
NodeList in                     ServiceAccountList in 
PersistentVolumeClaim in        Service in 
PersistentVolumeClaimList in    ServiceList in 
PersistentVolume in             Status in 
```

**Comment by mml** [comment]
@mikedanese did you already fix this?

**Comment by mikedanese** [comment]
Yup. Fixed in #20598
